src/anyio/_core/_synchronization.py

```diff
@@ -482,6 +520,95 @@ def statistics(self) -> CapacityLimiterStatistics:
         raise NotImplementedError
 
 
+class CapacityLimiterAdapter(CapacityLimiter):
+    _internal_limiter: CapacityLimiter | None = None
+
+    def __new__(cls, total_tokens: float) -> CapacityLimiterAdapter:
+        return object.__new__(cls)
+
+    def __init__(self, total_tokens: float) -> None:
+        if not isinstance(total_tokens, int) and total_tokens is not math.inf:
+            raise TypeError("total_tokens must be an int or math.inf")
+        elif total_tokens < 1:
+            raise ValueError("total_tokens must be >= 1")
+
+        self._total_tokens = total_tokens
+
+    @property
+    def _limiter(self) -> CapacityLimiter:
+        if self._internal_limiter is None:
+            self._internal_limiter = get_async_backend().create_capacity_limiter(
+                self._total_tokens
+            )
+
+        return self._internal_limiter
+
+    async def __aenter__(self) -> None:
+        await self._limiter.__aenter__()
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> bool | None:
+        return await self._limiter.__aexit__(exc_type, exc_val, exc_tb)
+
+    @property
+    def total_tokens(self) -> float:
+        if self._internal_limiter is None:
+            return self._total_tokens
+
+        return self._internal_limiter.total_tokens
+
+    @total_tokens.setter
+    def total_tokens(self, value: float) -> None:
+        self._limiter.total_tokens = value
```

Comment:
Frankly, the idea that someone would change the adapter's total tokens from a worker thread had not even occurred to me. The whole idea seems pretty bizarre to me.

Reply:
I'll revert my latest commit then?

Reply:
I have some mixed feelings about this. It might be reasonable to allow the tokens to be set from outside the event loop, just like `trio.CapacityLimiter` allows.

Reply:
The more I think about it, the more I'm leaning towards allowing users to mutate `total_tokens` before the initialization.

Reply:
Yeah, generally the API should be considered thread-unsafe, save for that parts that were explicitly designed to allow access from worker threads (the `anyio.from_thread` module).